### DISCUSSION TITLE
Primary navigation

### DISCUSSION THREAD

**Discussion by robertjmccarthy** [ministryofjustice/moj-frontend#710] [RELEASED: Components, pages and patterns]
The [primary navigation component](https://design-patterns.service.justice.gov.uk/components/primary-navigation/) is live in the MoJ Design System.

Use this discussion to suggest changes and improvements to this component.

**Comment by aliuk2012** [comment]
Currently, this component does not support non-link navigation. For example, if the user is on the current page of the navigation there is no need for the navigation item to be a link instead it should be place text. Expected it to be bit like breadcrumb nav where the last item in the list is never a link.  

A possible fix using the existing macro options would be if the `active` attribute is `true` then the component could return on the text and not bother using the href.

**Reply by robertjmccarthy** [reply]
Thanks @aliuk2012 for the suggested improvement.

Most likely one for @chrispymm but if I have understood the suggestion correctly, could there be an accessibility issue with this change? Looking at other nav components across the MoJ Design System and GOV Design System, active states are shown with hover and focus. If the link was no longer a link and instead text, none of these states would be present.

Apologies if I haven't understood your suggestion correctly.

**Other nav components**
- https://design-patterns.service.justice.gov.uk/components/sub-navigation/
- https://design-patterns.service.justice.gov.uk/components/side-navigation/
- https://design-patterns.service.justice.gov.uk/components/sub-navigation/
- https://design-system.service.gov.uk/components/service-navigation/

**Reply by aliuk2012** [reply]
Ah yes, that is true. Its not an accessibility concern because  active states would help the user. I was comparing it to breadcrumbs which currently we directly under the navigation and how the last item in the breadcrumb is never a link because the user is already on the page. The difference with breadcrumbs is the navigation is used to jump to previous pages where a primary navigation could be used at any point.

I had a quick review through GOV.UK's Service Navigation component nunjucks options and it looks like they do provide options for non-link text.
https://design-system.service.gov.uk/components/service-navigation/#options-service-navigation-example
![image](https://github.com/user-attachments/assets/4c0013f9-4fa5-4495-98cb-a7d0515feb21)


I noticed that `href` attribute was not required and thought that might have been a typo but looking at the code it would seem that they do support non-link text in the navigation https://github.com/Laings-Online-Limited/govuk-frontend/blob/1131be1121064325bda84a0b919b8e809d468a80/packages/govuk-frontend/src/govuk/components/service-navigation/template.njk#L63-L74. I've tried digging around the documentation they have to find out what the user case would be for non-link text or any finders with accessibility issues but I have not found any.
